geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/BoundsFilterVisitor.scala

```diff
@@ -0,0 +1,83 @@
+      case f: And                            => visitor.visit(f, envelope)
+      case f: Id                             => visitor.visit(f, envelope)
+      case f: Not                            => visitor.visit(f, envelope)
+      case f: Or                             => visitor.visit(f, envelope)
+      case f: PropertyIsBetween              => visitor.visit(f, envelope)
+      case f: PropertyIsEqualTo              => visitor.visit(f, envelope)
+      case f: PropertyIsNotEqualTo           => visitor.visit(f, envelope)
+      case f: PropertyIsGreaterThan          => visitor.visit(f, envelope)
+      case f: PropertyIsGreaterThanOrEqualTo => visitor.visit(f, envelope)
+      case f: PropertyIsLessThan             => visitor.visit(f, envelope)
+      case f: PropertyIsLessThanOrEqualTo    => visitor.visit(f, envelope)
+      case f: PropertyIsLike                 => visitor.visit(f, envelope)
+      case f: PropertyIsNull                 => visitor.visit(f, envelope)
+      case f: PropertyIsNil                  => visitor.visit(f, envelope)
+      case f: BBOX                           => visitor.visit(f, envelope)
+      case f: Beyond                         => visitor.visit(f, envelope)
+      case f: Contains                       => visitor.visit(f, envelope)
+      case f: Crosses                        => visitor.visit(f, envelope)
+      case f: Disjoint                       => visitor.visit(f, envelope)
+      case f: DWithin                        => visitor.visit(f, envelope)
+      case f: Equals                         => visitor.visit(f, envelope)
+      case f: Intersects                     => visitor.visit(f, envelope)
+      case f: Overlaps                       => visitor.visit(f, envelope)
+      case f: Touches                        => visitor.visit(f, envelope)
+      case f: Within                         => visitor.visit(f, envelope)
+      case f: After                          => visitor.visit(f, envelope)
+      case f: AnyInteracts                   => visitor.visit(f, envelope)
+      case f: Before                         => visitor.visit(f, envelope)
+      case f: Begins                         => visitor.visit(f, envelope)
+      case f: BegunBy                        => visitor.visit(f, envelope)
+      case f: During                         => visitor.visit(f, envelope)
+      case f: EndedBy                        => visitor.visit(f, envelope)
+      case f: Ends                           => visitor.visit(f, envelope)
+      case f: Meets                          => visitor.visit(f, envelope)
+      case f: MetBy                          => visitor.visit(f, envelope)
+      case f: OverlappedBy                   => visitor.visit(f, envelope)
+      case f: TContains                      => visitor.visit(f, envelope)
+      case f: TEquals                        => visitor.visit(f, envelope)
+      case f: TOverlaps                      => visitor.visit(f, envelope)
```

Comment:
There's a comment at the top - basically the geotools bounds visitor for some reason doesn't accept a generic filter :/